### PR TITLE
Commandlineoptions

### PR DESCRIPTION
**Pull request by rajrao**
#416 Document the commandline options that ServiceBusExplorer supports

if run with serviceBusExplorer.exe --help:
image

And adds a toolbar option.
image

utilizes https://github.com/commandlineparser (https://www.nuget.org/packages/CommandLineParser/)

### CODE REVIEW

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-commandlineoptions.51.build.627 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30298438) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/0e047f301a by @)

File: src/ServiceBusExplorer.sln.DotSettings [hunk 1]
```diff
@@ -590,6 +590,9 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FFIELD/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/XamlNaming/UserRules/=XAML_005FRESOURCE/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
 	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpFileLayoutPatternsUpgrade/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer.sln.DotSettings]: Is this accidental change?

File: src/ServiceBusExplorer/ServiceBusExplorer.csproj [hunk 2]
```diff
@@ -587,6 +588,9 @@
     <WCFMetadata Include="Service References\" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="CommandLineParser">
```
Comment (SeanFeldman) [inline src/ServiceBusExplorer/ServiceBusExplorer.csproj]: For consistency, could you please combine version  with package reference? Thanks 🙂

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-commandlineoptions.52.build.628 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30300241) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/61cad8af65 by @rajrao)

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-commandlineoptions.53.build.629 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30300313) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/25c52ef930 by @rajrao)

File: src/ServiceBusExplorer.sln.DotSettings (hunk 1 above)
Comment (rajrao) [reply src/ServiceBusExplorer.sln.DotSettings]: Done by Resharper in VS2019. Maybe caused by newer version? I have removed those lines

File: src/ServiceBusExplorer/ServiceBusExplorer.csproj (hunk 2 above)
Comment (rajrao) [reply src/ServiceBusExplorer/ServiceBusExplorer.csproj]: Done

File: src/ServiceBusExplorer.sln.DotSettings (hunk 1 above)
Comment (rajrao) [reply src/ServiceBusExplorer.sln.DotSettings]: Who is supposed to resolve the conversation? me or you @SeanFeldman ?

Comment (SeanFeldman) [reply src/ServiceBusExplorer.sln.DotSettings]: Don't worry about those 🙂

File: src/ServiceBusExplorer/CommandLineOptions.cs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    using CommandLine;
+    using CommandLine.Text;
+    using Helpers;
+
+    public class CommandLineOptions
+    {
+        [Option('n', "namespace", SetName = "Connection", Required = false, HelpText = "Load namespace key in the configuration file.")]
+        public string Namespace { get; set; }
+
+        [Option('c', "connectionString", SetName = "Connection", Required = false, HelpText = "use a connection string.")]
```
Comment (ErikMogensen) [inline src/ServiceBusExplorer/CommandLineOptions.cs]: "use -> "Use

Comment (rajrao) [reply src/ServiceBusExplorer/CommandLineOptions.cs]: Done!

**Comment by SeanFeldman** [comment]
:white_check_mark: [Build ServiceBusExplorer 4.2.0-commandlineoptions.54.build.630 completed](https://ci.appveyor.com/project/seanfeldman/servicebusexplorer/builds/30321692) (commit https://github.com/paolosalvatori/ServiceBusExplorer/commit/dfe334fa78 by @)

**Comment by rajrao** [comment]
Do I need to close this PR? Or is that done by you guys?

**Comment by SeanFeldman** [comment]
Thank you @rajrao. We just needed to merge it 🙂
All good now.

**Comment by paolosalvatori** [comment]
Thanks @rajrao for the great addition

**Pull request merged**